README.md

```diff
@@ -1,80 +1,179 @@
 
-Run/test your app by browsing to
+DNS resolution for a host name is handled separately from routing; your administrator may have configured a cloud domain that will always correctly resolve to the OpenShift router, or if using an unrelated host name you may need to modify its DNS records independently to resolve to the router.
 
-        $ curl service-ip-address:8080
+As of OpenShift v3.1 routes can be configured in the web console or via CLI. Using the CLI, you can expose a route as follows:
 
-#### update ####
+        $ oc expose service/<name> --hostname=<www.example.com>
 
-Assuming you used the URL of your own forked report, we can easily push changes to that hosted repo and simply repeat the steps above to build which will trigger the new built image to be deployed.
+If you're running OpenShift on a local machine, you can preview the new app by setting a local route like:
 
-#### delete ####
+        $ oc expose service/nodejs-ex --hostname=10.2.2.2
 
-		$ oc delete all -l name=myapp
+#### Create a new app from an image (method 3) 
 
-To remove all the resources with the label "name=myapp".
+You may have noticed the home page "Page view count" reads "No database configured". Let's fix that by adding a MongoDB service. We could use the second OpenShift template example (`nodejs-mongodb.json`) but for the sake of demonstration let's point `oc new-app` at a DockerHub image:
 
-###Debugging Unexpected Failures
+        $ oc new-app centos/mongodb-26-centos7 \
+        $ -e MONGODB_USER=admin,MONGODB_DATABASE=mongo_db,MONGODB_PASSWORD=secret,MONGODB_ADMIN_PASSWORD=super-secret
+        
+The `-e` flag sets the environment variables we want used in the configuration of our new app.
 
-Review some of the common tips and suggestions [here](https://github.com/openshift/origin/blob/master/docs/debugging-openshift.md).
+Running `oc status` will reveal the address of the newly created MongoDB:
+
+        In project nodejs (nodejs-echo) on server https://10.2.2.2:8443
+
+        http://10.2.2.2 to pod port 8080-tcp (svc/nodejs-ex)
+          dc/nodejs-ex deploys istag/nodejs-ex:latest <-
+            bc/nodejs-ex builds https://github.com/openshift/nodejs-ex with openshift/nodejs:0.10
+            deployment #1 deployed 12 minutes ago - 1 pod
+
+        svc/mongodb-26-centos7 - 172.30.201.71:27017
+          dc/mongodb-26-centos7 deploys istag/mongodb-26-centos7:latest
+            deployment #1 deployed 2 minutes ago - 1 pod
+
+
+Remove all the nodejs-ex resources with the label "name=myapp".
+
+        $ oc delete all -l name=myapp
 
+Recreate the nodejs service again, but this time specifying the MONGO_URL environment variable:
 
-### Web UI ###
+        $ oc new-app https://github.com/openshift/nodejs-ex -l name=myapp \
+        $ -e MONGO_URL=mongodb://admin:secret@172.30.201.71:27017/mongo_db
+
+#### Success
+
+This example will serve a welcome page and the current hit count as stored in a database to [http://10.2.2.2](http://10.2.2.2).
```

Comment:
i'm not sure where this 10.2.2.2 value came from (I see something similar appears to have been in the original readme), but it's not accurate since that's pointing to the openshift web console.  I'd just end the sentence as "stored in a database."